### PR TITLE
Add AMD Gluon sampling argmax

### PR DESCRIPTION
**Pull request by antiagainst**
Adding Gluon argmax kernels for AMD gfx950. Measured with

- Input shape: (M, N)
- Argmax dimension: N, the vocab dimension
- Dtypes: float32, float16, bfloat16
- M cases: 1, 2, 3, 4, 8, 16, 32, 64, 128
- Timing: triton do_bench(..., warmup=25, rep=100, quantiles=[0.5]) for both gluon and torch
- Table values are speedup: torch_us / gluon_us, so higher is better.

Overall geomean speedup across all 150 cases: 2.33x.
Detailed per model, average over precisions.

| Model | M=1 | M=2 | M=3 | M=4 | M=8 | M=16 | M=32 | M=64 | M=128 |
|---|---:|---:|---:|---:|---:|---:|---:|---:|---:|
| deepseek_v4 [vocab=129280] | 7.395x | 7.072x | 6.687x | 6.643x | 5.530x | 4.687x | 3.715x | 2.708x | 1.577x |
| qwen3_5 [vocab=151936] | 2.252x | 2.256x | 2.226x | 2.279x | 2.139x | 2.301x | 1.961x | 1.532x | 1.629x |
| kimi_k2_5 [vocab=163840] | 2.243x | 2.300x | 2.282x | 2.314x | 2.097x | 2.297x | 1.991x | 1.430x | 1.611x |
| minimax_m2 [vocab=200064] | 2.136x | 2.057x | 2.038x | 2.107x | 2.124x | 2.167x | 1.722x | 1.585x | 1.255x |
| gpt_oss [vocab=201088] | 2.126x | 2.071x | 2.046x | 2.114x | 2.122x | 2.158x | 1.728x | 1.578x | 1.312x |

### CODE REVIEW

File: tokenspeed-kernel/python/tokenspeed_kernel/ops/sampling/gluon/argmax_gfx950.py [hunk 1]
```diff
@@ -0,0 +1,426 @@
...
+    vals = cdna4.buffer_load(
+        logits,
+        row * stride_m + cols,
+        mask=mask,
+        other=-float("inf"),
+    ).to(gl.float32)
+    if SANITIZE_NAN:
+        mask = mask & (vals == vals)
+        vals = gl.where(mask, vals, -float("inf"))
+    indices = gl.where(mask, cols.to(gl.int32), 2147483647)
```
Comment (borontion) [inline tokenspeed-kernel/python/tokenspeed_kernel/ops/sampling/gluon/argmax_gfx950.py]: 2147483647 better to be  `2^31 - 1`?

File: tokenspeed-kernel/python/tokenspeed_kernel/ops/sampling/gluon/argmax_gfx950.py [hunk 2]
```diff
@@ -0,0 +1,426 @@
...
+    if logits.stride(1) != 1:
+        return False
+    return True
+
+
+def _get_atomic_scratch(
+    M: int, num_splits: int, device: torch.device
+) -> tuple[torch.Tensor, torch.Tensor, torch.Tensor]:
+    device_index = device.index
+    if device_index is None:
```
Comment (borontion) [inline tokenspeed-kernel/python/tokenspeed_kernel/ops/sampling/gluon/argmax_gfx950.py]: this `if device_index is None` should be redundant? 

given `device` comes from `logits.device`, so it should always be valid?

File: tokenspeed-kernel/python/tokenspeed_kernel/ops/sampling/gluon/argmax_gfx950.py (hunk 1 above)
Comment (antiagainst) [reply tokenspeed-kernel/python/tokenspeed_kernel/ops/sampling/gluon/argmax_gfx950.py]: Done

File: tokenspeed-kernel/python/tokenspeed_kernel/ops/sampling/gluon/argmax_gfx950.py (hunk 2 above)
Comment (antiagainst) [reply tokenspeed-kernel/python/tokenspeed_kernel/ops/sampling/gluon/argmax_gfx950.py]: Done

**Review by borontion — APPROVED** [review]
